Scripts/RemediationScripts/Remediate-ConfigureAADAuthExtOnLinuxVMSS.ps1

```diff
@@ -0,0 +1,1109 @@
+            # Get-AzVmss -ResourceGroupName V-HARARORATESTRG -VMScaleSetName testvmssForBRS1 -InstanceView: $false -UserData
+            $VMSSDetails | ForEach-Object{
+                $VMSS = $_
+                $VMSS | Add-Member -NotePropertyName isExtPresent -NotePropertyValue $false
+                if ($VMSS.OrchestrationMode -eq "Uniform" -and $VMSS.OsType -eq "Linux")
+                {
+                    $VMSS.Extensions | ForEach-Object{
+                        $VMSSExtension= $_
+                        if($VMSSExtension.Type -eq $reqExtensionType -and $VMSSExtension.Publisher -eq $reqExtPublisher)
+                        {
+                            $VMSS.isExtPresent = $true
+                        }
+                    }
+                    if(!$VMSS.isExtPresent)
+                    {
+                        Write-Host "AAD Extension is not present in Virtual Machine Scale Set [$($VMSS.ResourceName)]. Adding..." -ForegroundColor $([Constants]::MessageType.Warning)
+                        Write-Host $([Constants]::SingleDashLine)
+                        $NonCompliantVMSSDetails += $VMSS
+                    }
+                    
+                }
+                else
+                {
+                    if($VMSS.OrchestrationMode -eq "Flexible")
+                    {
+                        Write-Host "Virtual Machine Scale Set [$($VMSS.ResourceName)] Orchestration type is Flexible. Skipping..." -ForegroundColor $([Constants]::MessageType.Warning)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                    else
+                    {
+                        Write-Host "Virtual Machine Scale Set [$($VMSS.ResourceName)] OS type is Windows. Skipping..." -ForegroundColor $([Constants]::MessageType.Warning)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                }
+            }
+
+        }
+        catch
+        {
+            Write-Host "Error fetching Virtual Machine Scale Set(s) from the subscription. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+        }    
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+        Write-Host "Fetching all Virtual Machine Scale Set(s) from [$($FilePath)]..." 
+
+        $VirtualMachineResources = Import-Csv -LiteralPath $FilePath
+        $validVirtualMachineResources = $VirtualMachineResources| Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+      
+        $validVirtualMachineResources| ForEach-Object {
+        $resourceId = $_.ResourceId
+            try
+            {
+                $VirtualMachineResource =  Get-AzVM -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
```

Comment:
Get-AzVM are we fethcing VM or VMSS?

Reply:
We are fetching VMSS, there was a typo I have fixed that.